presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -278,6 +295,46 @@ private static int getQueriesQueuedOnInternal(InternalResourceGroup resourceGrou
         return queriesQueuedInternal;
     }
 
+    @Managed
+    public int getTaskLimitExceeded()
+    {
+        return taskLimitExceeded.get() ? 1 : 0;
+    }
+
+    private int calculateRunningTaskCount()
+    {
+        int taskCount = 0;
+        for (RootInternalResourceGroup rootGroup : rootGroups) {
+            synchronized (rootGroup) {
+                taskCount += calculateRunningTaskCount(rootGroup);
+            }
+        }
+        return taskCount;
+    }
+
+    private int calculateRunningTaskCount(InternalResourceGroup resourceGroup)
```

Comment:
This might conflict with what @mayankgarg1990 was suggesting, but for consistency with how we compute other stats I think we could move this into `InternalResourceGroup`

Reply:
If we are moving num running tasks to be a stat like `cpuUsageMillis` then yes, lets move it in to the `InternalResourceGroup` class. If we want to have some kind of Supplier which indicates to pause scheduling to `InternalResourceGroupManager` then lets just operate at the root level. So I am fine either ways - just not with mixing them together

Reply:
Moved to `InternalResourceGroup`